### PR TITLE
QR Code Generator App

### PR DESCRIPTION
**Pull request by Jayprakash-SE**
QR Code Generator App to generate QR Codes.

### CODE REVIEW

File: apps/qrcode-generator/static/style.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#urlPara {
```
Comment (srish) [inline apps/qrcode-generator/static/style.css]: Consider renaming ids in the CSS file; short, all letters lower case and use of `-` to join words.

File: apps/qrcode-generator/templates/upload.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+{% block main %}
+
+{% if error is not none %}
+<div class="alert alert-danger" role="alert">
+  Something went wrong!
+</div>
+{% else %}
+<div class="alert alert-success" role="alert">
+  QR Code successfully uploaded on the commons
```
Comment (srish) [inline apps/qrcode-generator/templates/upload.html]: Rephrase this to: "Your QR code is successfully uploaded to Wikimedia commons!"

File: apps/qrcode-generator/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# QR Code Generator
+This is a qrcode generator for Wikimedia Site. This app also demonstrate the usage of MediaWiki API with [OAuth](https://www.mediawiki.org/wiki/Help:OAuth).
+
+Install
+-------
+
+```
+$ git clone https://github.com/Jayprakash-SE/qrcode-generator.git
+$ pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (Jayprakash-SE) [reply apps/qrcode-generator/README.md]: Done

[the remark this answers is not in the document]
Comment (Jayprakash-SE) [reply apps/qrcode-generator/README.md]: Done

[the remark this answers is not in the document]
Comment (Jayprakash-SE) [reply apps/qrcode-generator/README.md]: Done

File: apps/qrcode-generator/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# QR Code Generator
+This is a qrcode generator for Wikimedia Site. This app also demonstrate the usage of MediaWiki API with [OAuth](https://www.mediawiki.org/wiki/Help:OAuth).
```
[the remark this answers is not in the document]
Comment (Jayprakash-SE) [reply apps/qrcode-generator/README.md]: Done

File: apps/qrcode-generator/static/style.css (hunk 1 above)
Comment (Jayprakash-SE) [reply apps/qrcode-generator/static/style.css]: Done

File: apps/qrcode-generator/templates/upload.html (hunk 2 above)
Comment (Jayprakash-SE) [reply apps/qrcode-generator/templates/upload.html]: Done

**Comment by Jayprakash-SE** [comment]
> Overall, the app is looking great! One important thing, run your Python code through Pylint and HTML code as well through a linter (I usually use this one: https://pypi.org/project/html-linter/).

I have run app.py in Pylint. It is giving me 9/10. The only docstring is missing. And in HTML, There are some Info Messages to remove optional tag like head, body, and li, etc.

> * [x]  Disallow the app to accept/generate QR code for any input that is not in URL format
> * [x]  Remove the label "URL:" and "The URL must be of any Wikimedia project." Instead, add a placeholder text in the input field: "Enter a URL of an article page from any Wikimedia site."
> * [x]  Left align the QR code and submit button.
> * [x]  Use the same font for the buttons (Submit, Download, etc.) that you are using for other elements (headers and paragraphs).
> * [x]  Add an alert on top before the URL input (like this https://getbootstrap.com/docs/4.0/components/alerts/) that says: "Log in to the app with your MediaWiki account to upload the generated QR Code to Wikimedia Commons".
> * [x]  Place the Download and Upload to Commons button in the same row, next to each other.
> * [x]  Logout UI is ideally a combination of label + button, placed next to each other. In the label show: "You're logged in as username!" and in the button just the fa-user icon and text "Logout".
> * [x]  Rephrase the success alert to "Your QR code is successfully uploaded to Wikimedia Commons! Click here to view it." Link the, "Click here to view" sentence from the file URL on Commons.
> * [x]   Remove the "Link to Commons" button and make Go back button's color as blue and the text in it as "Go back to home page".
> * [x]  Change the text "Something went wrong" to "Something went wrong while uploading your QR code :-/"
> * [x]  Change the text in "Upload to commons" to "Start upload" and add fa-upload icon.
> * [x]  Change the text in the current "Upload" button to "Finish upload" and add fa-upload icon.
> * [x]  Keep the color of all buttons blue except login, logout and submit (keep these three same as submit).
> * [x]  Change the following text: "Upload the QR Code to Wikimedia Commons" to "Modify the title and add a description for QR code file:"

**Comment by Jayprakash-SE** [comment]
> * [ ]  Change the login/logout button's background to white and make sure on hover the text color of the button changes from white to grey or something that makes the text visible. Making this change shouldn't change the style of the "Submit" or any other button.

I have changed the class to btn-outline-light. Now it is more visible and As I have a little bit of experience of UI (Through the OOUI), So per UI, Button's background* to white will not be good. Because both are the opposite color. See new changes, If you want above change then otherwise I will amend the pull request. But transparent background with a white outline will be very good :)

> * [x]  Clicking the article link in "QRCode for Wiki article:  " should open the page in a different tab and not the same one.
> * [x]  There are some whitespaces in `index.html` as shown by `html_lint.py`. Would be cool to get rid of them.
> * [x]  The SVG image and the button rows in both pages are not aligned with the rest of the elements on the page. See red circle in screenshot:
